src/common.rs

```diff
@@ -1123,6 +1123,35 @@ pub fn handler_invalid_tx_error(
     }
 }
 
+fn get_near_price(api: Option<url::Url>) -> color_eyre::eyre::Result<Option<f64>> {
+    let url = if let Some(api) = api {
+        api.join("api/v3/simple/price?ids=near&vs_currencies=usd")?
+    } else {
+        return Ok(None);
+    };
+
+    for _ in 0..10 {
+        let response = reqwest::blocking::get(url.clone())?;
+
+        let body = response.text()?;
+        let parsed_body: serde_json::Value = serde_json::from_str(&body)?;
+
+        let price = parsed_body["near"]["usd"].as_f64();
+
+        if let Some(price) = price {
+            return Ok(Some(price));
+        }
+
+        std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
This implementation is not going to retry. If request will fail, it will return Err on `?`. Fix the implementation.